released/packages/coq-gappa/coq-gappa.1.4.4/opam

```diff
@@ -17,6 +17,7 @@ depends: [
   "ocaml"
   "coq" {>= "8.8.1" & < "8.13~"}
   "coq-flocq" {>= "3.0"}
+  "gappa" {>= "1.3.5"}
```

Comment:
Please do not add this dependency. The presence of `gappa` is not needed to build `coq-gappa`, and updating `gappa` should not cause a recompilation of `coq-gappa`. (So, it is not even a `depopts`.) If you need to express the relationship between versions, use instead
```
conflicts: [
  "gappa" {< "1.3" }
]
```
but that would not bring much (by lack of such packages).

Reply:
I thought about this. The reason why I added the dependency is that the plugin is not usable without gappa. So while it is not a build dependency, it is a runt time dependency. I think that users can reasonably expect that coq-gappa does install the runtime dependencies it needs. But I agree that it is a disadvantage for CI, but only if CI does not test if the resulting plugin actually works - which I think CI should do.

Reply:
We could certainly change `build` to
```
build: [
  ["autoconf"] {dev}
  ["./configure"]
  ["./remake" "-j%{jobs}%"]
  ["./remake" "check"] {with-test}
]
```
and then it would make sense to have
```
depends: [
  "gappa" {with-test}
]
```

But that is an orthogonal matter. (And does the CI actually call `opam install --with-test`?)

Reply:
good point: it might be useful to run tests as well in opam-coq-archive.
FWIW, if tweaking the CLI command itself is cumbersome, it suffices to add the environment variable
```
OPAMWITHTEST: "true"
```
(we actually noticed this with @Zimmi48 when we recently added support for [environment variables in docker-coq-action](https://github.com/coq-community/docker-coq-action#export))

Reply:
@silene : My point is that the main purpose of the opam packages is installation by users and not convenience in test systems. I don't see a situation where a user would like to install coq-gappa and not gappa. And a user is also not likely to explicitly request tests. If we want to make it such that in tests coq-gappa compiles without gappa for CI performance reasons, then it should be done such that it is explicitly removed in CI testing. Not sure how one would do that - I guess we would need some quick-test flag.

Reply:
> updating `gappa` should not cause a recompilation of `coq-gappa`.

This is a legitimate concern though. It seems that opam supports this use case with `post` dependencies (run-time only dependencies IIUC): see https://opam.ocaml.org/doc/Manual.html#opamfield-depends.

Reply:
Unfortunately, that is not how `post` works (which may explain why almost no package uses it, except OCaml itself, for which it was designed). When running Opam, there are two resolution stages. First, the set of needed actions (remove, install) is computed by calling a constraint solver. Second, the directed acyclic graph of needed actions is computed. The `post` variable is only used during this second stage, to decide whether to put some edges between vertices. So, as soon as you state that `gappa-coq` depends on `gappa` (whether `post` of not), the remove/install vertices for `gappa-coq` will be part of the graph when modifying `gappa`, since they have been found during the first resolution stage. The `post` part just means that there is no edge between the vertices for installing `gappa` and `gappa-coq`. So, their installation will happen in a random order.

Reply:
> updating gappa should not cause a recompilation of coq-gappa.

Compared to gappa, coq-gappa is quite light weight, so I don't see much of an issue with that. It is also not so that gappa changes every other week. The main issue I see is that people install coq-gappa and then it doesn't work.

So if the options are:
1.) people have to know that if they want to use gappa in Coq, they have to install gappa and coq-gappa
2.) coq-gappa is recompiled without need when gappa changes

I am for option 2. You might want to read (https://www.npr.org/2019/03/08/701417140/when-not-reading-the-fine-print-can-cost-your-soul) to get an idea on how many people read a "please install gappa if you install coq-gappa" note before they find that coq-gappa does not work - especially when coq-gappa is the prerequisite of some higher level automation.